core-bundle/src/Controller/CspReporterController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\PageModel;
+use Nelmio\SecurityBundle\Controller\ContentSecurityPolicyController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Attribute\Route;
+
+#[Route(path: '/_contao/csp/report/{page}', methods: [Request::METHOD_POST])]
+final class CspReporterController
+{
+    public function __construct(
+        private readonly ContaoFramework $framework,
+        private readonly ContentSecurityPolicyController|null $nelmioController,
+    ) {
+    }
+
+    public function __invoke(Request $request, int $page): Response
+    {
+        if (!$this->nelmioController) {
```

Comment:
Imho it would be better to make `nelmioController` a mandatory argument (because it cannot work without it) and unregister the entire controller, if the bundle is not there. Or the other way around, only register if it is there.

Reply:
My thinking was that if the nelmio controller does not exist (in a non-managed edition for instance) then this way the route still works - but just returns 404.

If we don't want this then I think I'd opt for not creating our own controller at all but instead do the following:

* Add our own route using Nelmio's existing controller.
* Implement our own report filter that checks for the page parameter.
* Re-introduce a boolean parameter for the `ContaoResponseContextFactory` that disables the generation of the `report-uri` - otherwise the router might unnecessarily be asked for the non-existent route on every request.

On disadvantage to this approach would be, that the CSP report filter is not the first check in the request. Nelmio's CSP controller first parses the request content and if that fails a log entry will be generated (https://github.com/nelmio/NelmioSecurityBundle/blob/da7cc88ae338c6b8370c45c2e5fec88d24f5595d/src/Controller/ContentSecurityPolicyController.php#L40-L46). Since the controller's route will always exist for us then it would be possible to hammer the log here.

Reply:
If the route is not defined, you will also get a 404 anyway? I think having our controller is fine but it doesn't need to exist just to send a 404, does it?

Reply:
The route will always be defined (we define the route, not nelmio) - unless we use a dynamic route loader (which was my original implementation).

Reply:
I'm fine with this then :)